src/Microsoft.AspNetCore.Sockets.Http/HttpConnectionDispatcher.cs

```diff
@@ -488,15 +488,34 @@ private static void WriteTransport(JsonWriter writer, string transportName, Tran
                 return false;
             }
 
+            // Setup the connection state from the http context
+            connection.User = context.User;
+
             // Configure transport-specific features.
             if (transportType == TransportType.LongPolling)
             {
                 connection.Features.Set<IConnectionInherentKeepAliveFeature>(new ConnectionInherentKeepAliveFeature(options.LongPolling.PollTimeout));
-            }
 
-            // Setup the connection state from the http context
-            connection.User = context.User;
-            connection.SetHttpContext(context);
+                // For long polling, the requests come and go but the connection is still alive.
+                // to make the IHttpContextFeature work well, we make a copy of the relevant properties
+                // to a new HttpContext. This means that it's impossible to affect the context
+                // with subsequent requests.
+                var existing = connection.GetHttpContext();
+                if (existing == null)
+                {
+                    var httpContext = CloneHttpContext(context);
+                    connection.SetHttpContext(httpContext);
+                }
+                else
+                {
+                    // Set the request trace identifier to the current http request handling the poll
+                    existing.TraceIdentifier = context.TraceIdentifier;
```

Comment:
:+1:

Any other properties we want to update? Connection info (IP/Port/etc; In case a client changes networks, etc.)?

Of course, this is run in parallel with existing invocations so the data could "shear" (but as long as each set is atomic, it won't break anything).

Reply:
Extremely unlikely this would matter in real life. The reason I changed the trace identifier is because we log it to tell the user what the previous connection id is.